packages/app-project/src/components/AuthModal/components/LoginForm/LoginFormContainer.js

```diff
@@ -19,27 +28,28 @@ class LoginFormContainer extends Component {
     }
   }
 
-  onSubmit (values, { setFieldError, setSubmitting }) {
-    if (this.state.error) this.setState({ error: '' })
-    const { authClient, store } = this.props
-    return authClient.signIn(values)
-      .then(userResource => {
-        setSubmitting(false)
-        store.user.set(userResource)
-        this.props.closeModal()
-      })
-      .catch(error => {
-        console.error(error)
-        if (error && error.message === 'Invalid email or password.') {
-          const errorMessage = counterpart('LoginForm.error')
-          setFieldError('login', errorMessage)
-          setFieldError('password', errorMessage)
-        } else {
-          this.setState({ error: error.message })
-        }
+  async onSubmit (values, { setFieldError, setSubmitting }) {
+    if (this.state.error) {
+      this.setState({ error: '' })
+    }
 
-        setSubmitting(false)
-      })
+    try {
+      const { authClient, userStore } = this.props
+      const userResource = await authClient.signIn(values)
+      userStore.set(userResource)
+      this.props.closeModal()
+      setSubmitting(false)
+    } catch (error) {
+      console.error(error)
+      if (error && error.message === 'Invalid email or password.') {
```

Comment:
String matching shouldn't be used to detect the type of error. This should either use the response status code or the client should throw a custom error for this case.